### PR TITLE
Added GAD channel support.

### PR DESCRIPTION
**Pull request by MichaelBroughton**

### CODE REVIEW

File: tensorflow_quantum/core/src/circuit_parser_qsim.cc [hunk 1]
```diff
@@ -641,6 +641,25 @@ inline Status DepolarizingChannel(const Operation& op,
...
 
+inline Status GADChannel(const Operation& op, const unsigned int num_qubits,
+                         const unsigned int time, NoisyQsimCircuit* ncircuit) {
+  int q;
+  bool unused;
+  float p, gamma;
+  Status u;
+  unused = absl::SimpleAtoi(op.qubits(0).id(), &q);
+
+  u = ParseProtoArg(op, "p", {}, &p);
```
Comment (zaqqwerty) [inline tensorflow_quantum/core/src/circuit_parser_qsim.cc:652]: Should have two separate statuses here, currently bad status from parsing p will be ignored

**Review by zaqqwerty — CHANGES_REQUESTED** [review]
Error handling nit

File: tensorflow_quantum/core/src/circuit_parser_qsim.cc (hunk 1 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/src/circuit_parser_qsim.cc:652]: Good catch. Made consistent with other gates.

**Review by zaqqwerty — APPROVED** [review]
LGTM

**Pull request merged**